### ISSUE TITLE
Programmatically set filter?

### ISSUE THREAD

**Issue by jmreicha** [transcend-io/terragrunt-atlantis-config#228]
This is basically a question, wasn’t sure where else to post.

We have a large Terragrunt infrastructure and so running the generate command is causing high cpu and disk io. My first thought to help with this was to use filter to shrink down the number of directories that need to be parsed but haven’t seen a reliable way to get the top level directory to parse programmatically through any of the Atlantis env vars or other methods so that when the pre workflow runs it grabs only the environment it needs.

Any guidance on how this could be achieved with the least amount of intervention?

**Comment by dmattia** [comment]
Hey @jmreicha! Outside of `--filter` I'm not sure what other options would make the most sense, other than maybe introducing a maximum amount of concurrency that this library takes on at a time.

Was the high CPU/IO causing problems on your Atlantis server?

**Comment by jmreicha** [comment]
> Was the high CPU/IO causing problems on your Atlantis server?

It was yeah. We’re running Atlantis in Fargate so are somewhat limited on CPU.

**Comment by dmattia** [comment]
Makes sense. I think having a maximum concurrency maxes sense for this if you agree

**Comment by jmreicha** [comment]
Yeah outside of some existing environment/workaround that I'm not aware of I'm not sure of another solution. I'd be happy to help test it out if it makes sense to implement.

**Comment by SerhiiSokolov** [comment]
Hi @dmattia 
Regarding the CPU usage. When I start `terragrunt-atlantis-config` on my Mac M1 with 8 cores it takes about 5 minutes to generate `atlantis.yaml` which has around 5000 lines. 
I use this command 
`terragrunt-atlantis-config generate --output atlantis.yaml --parallel --create-workspace --autoplan --filter infra/[^_]*`
Could you advise why it takes so long?
